test/ngAnimate/animateCssSpec.js

```diff
@@ -1203,6 +1203,47 @@ describe("ngAnimate $animateCss", function() {
       });
     });
 
+    it('should avoid applying the same cache to an element a follow-up animation is run on the same element',
+      inject(function($animateCss, $rootElement, $document) {
+
+      var element = jqLite('<div></div>');
+      element.attr('class', '');
+
+      $rootElement.append(element);
+      jqLite($document[0].body).append($rootElement);
+
+      var runner = $animateCss(element, {
+        duration: 0.5,
+        to: { background: 'red' }
+      }).start();
+
+      triggerAnimationStartFrame();
+      expect(element.attr('style')).toContain('transition');
+
+      browserTrigger(element, 'transitionend',
+        { timeStamp: Date.now(), elapsedTime: 0.5 });
+
+      expect(element.attr('style')).not.toContain('transition');
+
+      runner = $animateCss(element, {
+        duration: 0.8,
+        to: { background: 'blue' }
+      }).start();
+
+      triggerAnimationStartFrame();
+      expect(element.attr('style')).toContain('transition');
+
+      browserTrigger(element, 'transitionend',
+        { timeStamp: Date.now(), elapsedTime: 0.5 });
```

Comment:
It passes because the transition is still active. Only when `800ms` kicks in will the animation be over. Is this what you mean or did I miss something?